jina/executors/indexers/vector/numpy.py

```diff
@@ -53,7 +53,7 @@ def query(self, keys: np.ndarray, top_k: int, *args, **kwargs) -> Tuple['np.ndar
         elif self.metric == 'cosine':
             dist = _cosine(keys, self.query_handler)
 
-        idx = dist.argsort(axis=1)[:, :top_k]
+        idx = np.argpartition(dist, kth=top_k, axis=1)[:, :top_k]
```

Comment:
so i'm guessing this is faster?

Reply:
it should, but I have not yet empirically tested

Reply:
![image](https://user-images.githubusercontent.com/4329072/87312125-be5d5800-c552-11ea-9188-68245ca58250.png)
Reference: [https://stackoverflow.com/questions/10337533/a-fast-way-to-find-the-largest-n-elements-in-an-numpy-array](https://stackoverflow.com/questions/10337533/a-fast-way-to-find-the-largest-n-elements-in-an-numpy-array)